src/hooks/useYoutubeApi.js

```diff
@@ -1,23 +1,28 @@
-import {useState, useEffect} from 'react';
+import {useState, useEffect, useCallback} from 'react';
 import youtube from '../apis/youtube';
 
 const useYoutubeApi = (query) => {
   const [videos, setVideos] = useState([]);
 
-  const handleSubmit = async (search) => {
-    const response = await youtube.get('/search', {
+  const handleSubmit = useCallback(async (search) => {
+    
+    try{
+      const response = await youtube.get('/search', {
       params: {
         q: search,
         type: 'video' 
       }
-    })
+      })
+      setVideos(response.data.items);
+    } catch(e){
+      console.log(e);
+    }
     
-    setVideos(response.data.items);
-  }
+  }, [])
```

Comment:
Also here. I do think that most of the logic is the same in the hooks, consider creating a generic function to keep your hooks as DRYer (don't repeat yourself) as possible.